pkgs/games/rotp/default.nix

```diff
@@ -0,0 +1,53 @@
+{ lib, stdenv, fetchFromGitHub, makeWrapper, jdk, jre }:
+
+stdenv.mkDerivation {
+  pname = "remnants-of-the-precursors";
+  version = "1.04";
+
+  src = fetchFromGitHub {
+    owner = "rayfowler";
+    repo = "rotp-public";
+    rev = "e3726fc22c2c44316306c50c79779e3da1c4c140";
+    sha256 = "sha256-oMA8LRpBoBX7t4G+HuRz0a8g+UEwYO7Ya0Qq4j8AWec=";
+  };
+
+  nativeBuildInputs = [ jdk makeWrapper ];
+
+  # By default, the game tries to write to the java class path. If that fails
+  # (and it always does, since they are in the read-only nix store), it won't
+  # launch.
+  patches = [ ./0001-store-config-and-saves-in-XDG_CONFIG_HOME.patch ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    javac -d "$out/share/" -sourcepath src src/rotp/Rotp.java
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # We need the assets, but don't want the source files.
+    cp -r src/rotp/* "$out/share/rotp/"
+    find "$out/share/rotp" -type f -name '*java' -exec rm "{}" \;
```

Comment:
Can we do this the other way around? Cross device copy is slower than first deleting the files.

Reply:
That's a good point. I've switched the order of steps.